examples/chess/src/chess_rs/Cargo.toml

```diff
@@ -15,4 +15,5 @@ candid = "0.8.0"
 ic-cdk = { path = "../../../../src/ic-cdk" }
 ic-cdk-macros = { path = "../../../../src/ic-cdk-macros" }
 serde = "1.0.111"
-pleco = "0.5.0"
+tanton = "1.0.0"
+getrandom = { version = "0.2", features = ["js"] }
```

Comment:
Adding support for WebAssembly for getrandom ([Docs](https://docs.rs/getrandom/latest/getrandom/#webassembly-support))

Reply:
The canister wasm execution runtime is different from in a browser when it comes to randomness.
There is currently no synchronous randomness API exposed. The only way to get "true" randomness is calling [`raw_rand`](https://internetcomputer.org/docs/current/references/ic-interface-spec#ic-raw_rand) method of the management canister.

Adding `getrandom` here won't make a difference. I'll remove it.

Reply:
@lwshang For some reason, without `getrandom = { version = "0.2", features = ["js"] }`, this test fails:

https://github.com/dfinity/cdk-rs/actions/runs/3568798481/jobs/5998064054#step:9:85

Adding it fixes the test though...

Reply:
Yes.

IMO, this is an issue with `tanton` crate. It should have a feature flag to gate its use of `rand`.

I added it back to make compiler happy. While, I believe it will cause problem when using `tanton`'s random related API in canisters.

Reply:
Enable `js` feature of `getrandom` actually adds an import section in the wasm module which is rejected by the IC replica.

So we have to fix `tanton` crate as I said above before we can merge this PR.

Reply:
After more investigation, I believe this is something need to be fixed on `getrandom` side.
It should not forbid compilation without `js` feature enabled.

`pleco` depends on older `rand`/`rand_core` which doesn't introduce `getrandom` indirectly. That's why there is no issue on the main branch.

Reply:
Found that it is possible to register custom `getrandom` function as described [here](https://docs.rs/getrandom/latest/getrandom/macro.register_custom_getrandom.html).

I copied the code and made the example test pass.